great_expectations/data_context/data_context/context_factory.py

```diff
@@ -140,6 +141,57 @@ def get_validator(self, batch_request: BatchRequest) -> Validator:
             )
         return self._project.get_validator(batch_request=batch_request)
 
+    def is_using_cloud(self) -> bool:
+        if not self._project:
+            raise RuntimeError(
+                "This action requires an active DataContext. "
+                + "Please call `great_expectations.get_context()` first, then try your action again."
+            )
+
+        from great_expectations.data_context import CloudDataContext
+
+        return isinstance(self._project, CloudDataContext)
+
+    def build_data_docs(
+        self,
+        site_names: list[str] | None = None,
+        resource_identifiers: list | None = None,
+        dry_run: bool = False,
+        build_index: bool = True,
+    ) -> dict:
+        if not self._project:
+            raise RuntimeError(
+                "This action requires an active DataContext. "
+                + "Please call `great_expectations.get_context()` first, then try your action again."
+            )
+
+        return self._project.build_data_docs(
+            site_names=site_names,
+            resource_identifiers=resource_identifiers,
+            dry_run=dry_run,
+            build_index=build_index,
+        )
+
+    def get_docs_sites_urls(
+        self,
+        resource_identifier: Any | None = None,
+        site_name: str | None = None,
```

Comment:
I haaaaaate this but what's the alternative? I could encapsulate all data docs CRUD into a `DataDocsManager` class so we can inject that but no matter what, we're going to have some messiness here.

I'd vote to get this in as-is and follow up with that docs encapsulation